src/lumigo_opentelemetry/libs/sampling.py

```diff
@@ -72,107 +96,120 @@ def get_description(self) -> str:
-    if not filter_regex_string:
+
+    return any(does_match_regex_safe(regex, endpoint) for regex in regexes)
+
+
+def does_match_regex_safe(regex: str, value: str) -> bool:
+    if not regex or value is None:
         return False
+
     try:
-        filter_regex = re.compile(filter_regex_string)
+        return re.match(regex, value) is not None
     except Exception:
+        logger.warning(f"Invalid regex: {regex}")
+        return False
+
+
+def _get_string_list_from_env_var(env_var_name: str) -> List[str]:
+    env_var_value = os.environ.get(env_var_name)
+    if not env_var_value:
+        return []
+
+    parsed_list = None
+    try:
+        parsed_list = json.loads(env_var_value)
+    except Exception:
+        pass
+
+    if (
+        parsed_list is None
+        or not isinstance(parsed_list, list)
+        or any(not isinstance(item, str) for item in parsed_list)
+    ):
         logger.warning(
-            f"Invalid regex in '{AUTO_FILTER_HTTP_ENDPOINTS_REGEX}': {filter_regex_string}",
-            exc_info=True,
+            f"The value of env var '{env_var_name}' is not valid JSON format for an array of strings: {env_var_value}"
         )
-        return False
-    return filter_regex.search(url) is not None
+        return []
+
+    return parsed_list
+
+
+def _extract_endpoint(attributes: Attributes, spanKind: SpanKind) -> Optional[str]:
+    """
+    Extract the endpoint from the given span attributes.
+
+    Expected attributes for HTTP CLIENT spans:
+    * url.full - The Absolute URL describing a network resource. E.g. "https://www.foo.bar/search?q=OpenTelemetry#SemConv"
+
+    Expected attributes for HTTP SERVER spans:
+    * url.path - The URI path component. E.g. "/search"
+
+    Deprecated attributes (see https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/#deprecated-http-attributes):
+    * http.target - replaced by the url.path & url.query attributes. Example: "/search?q=OpenTelemetry#SemConv"
+    * http.url - replaced by the url.full attribute. Example: "https://www.foo.bar/search?q=OpenTelemetry#SemConv"
+    """
+    if attributes is None:
+        return None
+
+    endpoint = None
+    if spanKind == SpanKind.CLIENT:
+        endpoint = (
+            attributes.get("url.full")
+            or attributes.get("http.url")
+            or attributes.get("http.target")
+        )
+    elif spanKind == SpanKind.SERVER:
+        endpoint = (
+            attributes.get("url.path")
+            or attributes.get("http.target")
+            or attributes.get("http.url")
+        )
+
+    endpoint = str(endpoint) if endpoint else None
+
+    if not endpoint:
+        return None
+
+    parsed_endpoint: ParseResult = urlparse(endpoint)
+    if spanKind == SpanKind.CLIENT:
+        return parsed_endpoint.geturl()
+    elif spanKind == SpanKind.SERVER:
+        # Make sure that we only return the path and everything that comes after it, not the full URL
```

Comment:
Why? is it because the parts we keep are the only ones we want to match on? if so, I'd add it to the comment

Reply:
In server spans the endpoint we are matching against is in the format: `/foo?param1=value1` (the format of the otel url.path param).
In Client spans the endpoint we are matching against is in the format: `https://example.com/foo/bar?param1=value1` (the format of the otel url.full param).

We are parsing the endpoint because we use fallback otel params that don't act consistently in every instrumentation (django vs fastapi for example), so we parse the endpoint and make sure it is only the path if we are in a server span.